### PR TITLE
Make it even easier to run Postgres tests; implement substring and trim

### PR DESCRIPTION
**Pull request by mihaibudiu**

### CODE REVIEW

File: sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/backend/DBSPCompiler.java [hunk 1]
```diff
@@ -230,12 +228,24 @@ private void compileInternal(String statements, boolean many, @Nullable String c
             }
         } catch (SqlParseException e) {
             this.messages.reportError(e);
+            if (this.options.optimizerOptions.throwOnError) {
+                throw new RuntimeException(e);
```
Comment (lalithsuresh) [inline sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/backend/DBSPCompiler.java:232]: Why not throw the exception as is?

Comment (mihaibudiu) [reply sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/backend/DBSPCompiler.java:232]: then you have to add them to the method signature.
This is only used for testing, normally they are converted into messages.

Comment (lalithsuresh) [reply sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/backend/DBSPCompiler.java:232]: You can have custom unchecked error types too by extending RuntimeException.

Comment (mihaibudiu) [reply sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/backend/DBSPCompiler.java:232]: The main point of this path is to immediately catch an exception during testing and have a valid stack trace that includes the origin point so I can study it under the debugger. Normally throwOnError is not set.
From my experience having a rich set of exceptions makes code much harder to maintain. Even with these I am not totally happy.

**Review by lalithsuresh — APPROVED** [review]

**Pull request merged**